runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp

```diff
@@ -1,404 +1,318 @@
+    std::vector<hailo_detection_with_byte_mask_t> detections = convert_nms_with_byte_mask_buffer_to_detections(logits);
+    cv::Mat overlay = cv::Mat::zeros(height, width, CV_8UC3);
+
+    // if (frame.channels() == 3){
+    //     cv::cvtColor(frame, frame, cv::COLOR_RGB2BGR);
+    // }
+
+    // std::cout << "Frame size: " << frame.size() << std::endl;
+    // std::cout << "Overlay size: " << overlay.size() << std::endl;
+
+    for(const auto& detection : detections) {
+        int box_width = (detection.box.x_max - detection.box.x_min) * width + 1;
+        int box_height = (detection.box.y_max - detection.box.y_min) * height + 1;
+
+        if (box_width <= width && box_height <= height ) {
+            for (size_t i = 0; i < static_cast<size_t>(box_height); ++i) {
+                for (size_t j = 0; j < static_cast<size_t>(box_width); ++j) {
+                    auto cropped_mask_idx = static_cast<size_t>(i) * box_width + j;
+                    if (detection.mask[cropped_mask_idx]) {
+                        int overlayX = static_cast<int>(j) + static_cast<int>(detection.box.x_min * width);
+                        int overlayY = static_cast<int>(i) + static_cast<int>(detection.box.y_min * height);
+                        if (overlayX >= 0 && overlayX < width && overlayY >= 0 && overlayY < height) {
+                            overlay.at<cv::Vec3b>(overlayY, overlayX) = cv::Vec3b(30, 255, 255);
+                        }
+                    }
+                }
+            }
+        cv::rectangle(frame, cv::Rect(detection.box.x_min * width, detection.box.y_min * height, box_width, box_height), cv::Scalar(30, 255, 255), 2);
         }
-
-        features.emplace_back(feature);
-    }
-
-    std::vector<cv::Mat> frames((int)frame_count);
-
-    auto input_thread(std::async(write_all, std::ref(input_vstream[0]), video_path, std::ref(write_time_vec), std::ref(frames)));
-
-    // Create read threads
-    std::vector<std::future<hailo_status>> output_threads;
-    output_threads.reserve(output_vstreams_size);
-    for (size_t i = 0; i < output_vstreams_size; i++) {
-        output_threads.emplace_back(std::async(read_all<T>, std::ref(output_vstreams[i]), features[i], frame_count, std::ref(read_time_vec[i]))); 
     }
+    cv::addWeighted(frame, 1, overlay, 0.3, 0.0, frame);
+    overlay.release();
+    return frame;
+}
 
-    auto pp_thread(std::async(post_processing_all<T>, std::ref(features), frame_count, std::ref(postprocess_time), std::ref(frames), org_height, org_width));
+template <typename T> hailo_status read_all(OutputVStream &output, std::string &video_path, int video_height, int video_width,  int height, int width, int frame_count, std::vector<cv::Mat>& frames) {
+    std::vector<T> data(output.get_frame_size());
+    std::cout << "-I- Started read thread " << video_path << std::endl;
+    cv::VideoWriter video("./processed_video.mp4",cv::VideoWriter::fourcc('m','p','4','v'),30, cv::Size(video_width, video_height));
```

Comment:
Video should not be saved by default. Either have it commented out with a note in the README or have it operational with a flag (like in this example: https://github.com/hailo-ai/Hailo-Application-Code-Examples/tree/main/runtime/cpp/detection/yolov8/x86_64) with a note in the README